### ISSUE TITLE
Darkbands at overlapping region

### ISSUE THREAD

**Issue by roshan64** [tranleanh/Image-Panorama-Stitching#1]
Hi,It's very useful and interesting repo.I am having a problem my stitched panoramic image there is visible darkbands at overlapping region.Can you please help me out why this is happening and how to get rid of this.
Thanks

Attachment:
[https://i.ibb.co/sgsS2Lb/b9212e4b-c664-4481-8b09-3915d21267e0.jpg](https://i.ibb.co/sgsS2Lb/b9212e4b-c664-4481-8b09-3915d21267e0.jpg)

**Comment by tranleanh** [comment]
Hi Ratan,

It may happen when your input images are captured in different light conditions.
Make sure that the common regions of each two images are in the same light condition.

Hope it useful.

**Comment by roshan64** [comment]
> regions

Thanks for the quick reply,My images has same exposure overall the region.Still I am getting that dark bands.I dont what is the issue.I am attaching my input images.
Thanks.
Attachments:
[https://i.ibb.co/YfmQ6PV/3.jpg](https://i.ibb.co/YfmQ6PV/3.jpg)
[https://i.ibb.co/ZYb5z4y/2.jpg](https://i.ibb.co/ZYb5z4y/2.jpg)

**Comment by tranleanh** [comment]
All your images are captured in the same exposure, right. But the left and the right sides of the images seem like not similar. I guess your images are from a microscope, and that may cause the problem. This algorithm will work more properly in ordinary images.

**Comment by roshan64** [comment]
> 
> 
> All your images are captured in the same exposure, right. But the left and the right sides of the images seem like not similar. I guess your images are from a microscope, and that may cause the problem. This algorithm will work more properly in ordinary images.

Yeah,it is working fine for ordinary images,but my project is on microscopic images :(
Is there any way to remove those dark bands?

**Comment by tranleanh** [comment]
I am not sure about this. But I think you should try some pre-processing methods to make your image more homogeneous. Try with histogram equalization.

**Comment by roshan64** [comment]
> 
> 
> I am not sure about this. But I think you should try some pre-processing methods to make your image more homogeneous. Try with histogram equalization.

tried.But same result.

**Comment by tranleanh** [comment]
Waiting for someone successfully dealing with this thing here.

**Comment by BunnyIcon** [comment]
Hi, your repo is very helpful. I'm wondering how to stitch more pictures together. When I stitched 20 pictures, it turns out "failed". If I stitch these pictures one by one, like in a for loop, then the stitched pic cannot be stitched with another one. Do you have any idea about this? Thanks in advance.

**Comment by tranleanh** [comment]
hi BunnyIcon, did you try with [multi_image_pano.py](https://github.com/tranleanh/image-panorama-stitching/blob/master/multi_image_pano.py), you can see the command for running at my instruction there.
